tests/conftest.py

```diff
@@ -729,3 +747,315 @@ def dummy_climate_data(fixture_core_components):
     )
 
     return data
+
+
+# dummy climate data with different number of canopy layers
+@pytest.fixture
+def dummy_climate_data_ragged(fixture_core_components, fixture_empty_array):
```

Comment:
I wouldn't really describe this as using ragged arrays. Because we are padding with `np.nan` to a fixed number of canopy layers, the data are always proper square arrays. It would be truly ragged if we stored the canopy for each cell as a list containing arrays with data for only the 'true' canopy layers. 

This is picky - but I think it _could_ confuse code review if people expect properly ragged structures across cells (which we most definitely get with the community data!).

```suggestion
def dummy_climate_data_canopy_variable(fixture_core_components, fixture_empty_array):
```

And then similar changes in the other names.

Reply:
Ok, I changed it to `varying_canopy` because I think using the word `variable` here might be confusing.